src/cryptography/x509/extensions.py

```diff
@@ -1446,6 +1446,138 @@ def __repr__(self):
     nonce = utils.read_only_property("_nonce")
 
 
+@utils.register_interface(ExtensionType)
+class IssuingDistributionPoint(object):
+    oid = ExtensionOID.ISSUING_DISTRIBUTION_POINT
+
+    def __init__(self, only_contains_user_certs, only_contains_ca_certs,
+                 indirect_crl, only_contains_attribute_certs,
+                 only_some_reasons=None, full_name=None, relative_name=None):
+
+        if (
+            only_some_reasons and (
+                not isinstance(only_some_reasons, frozenset) or not all(
+                    isinstance(x, ReasonFlags) for x in only_some_reasons
+                )
+            )
+        ):
+            raise TypeError(
+                "only_some_reasons must be None or frozenset of ReasonFlags"
+            )
+
+        if only_some_reasons and (
+            ReasonFlags.unspecified in only_some_reasons or
+            ReasonFlags.remove_from_crl in only_some_reasons
+        ):
+            raise ValueError(
+                "unspecified and remove_from_crl are not valid reasons in an "
+                "IssuingDistributionPoint"
+            )
+
+        bools = [
+            only_contains_user_certs, only_contains_ca_certs,
+            indirect_crl, only_contains_attribute_certs
+        ]
+
+        if not all([isinstance(x, bool) for x in bools]):
```

Comment:
Drop the list-comprehension. Also I think this is easier to read if you just write out the `and`s